storage/Devices/BcacheImpl.cc

```diff
@@ -86,6 +97,32 @@ namespace storage
     }
 
 
+    void
+    Bcache::Impl::reassign_numbers(Devicegraph* devicegraph)
+    {
+	// not many bcache devices expected (<10), so lets just do trivial
+	// recursive algorithm. It finds free name and ensure that all devices
```

Comment:
trivial recursive :rofl: 
The important thing to mention is "tail-recursive", practically a loop, which is easy to see thanks to the `return`, but also making sure the recursion does terminate. Can we explain that it indeed does?

Reply:
The recursive implementation does not look so efficient anyway since it queries all bcaches from the devicegraph (get_all) every time.